qcodes/dataset/data_set.py

```diff
@@ -1693,6 +1703,93 @@ def _flush_data_to_database(self, block: bool = False) -> None:
+        extension = export_type.value
+        return f"{prefix}{self.run_id}.{extension}"
+
+    def _export_as_netcdf(self, path: str, prefix: str) -> str:
+        """Export data as netcdf to a given path with file prefix"""
+        file_name = self._export_file_name(
+            prefix=prefix, export_type=DataExportType.NETCDF)
+        file_path = os.path.join(path, file_name)
+        xarr_dataset = self.to_xarray_dataset()
+        xarr_dataset.to_netcdf(path=file_path)
+        return path
+
+    def _export_as_csv(self, path: str, prefix: str) -> str:
+        """Export data as csv to a given path with file prefix"""
+        file_name = self._export_file_name(
+            prefix=prefix, export_type=DataExportType.CSV)
+        self.write_data_to_text_file(path=path, single_file=True, single_file_name=file_name)
+        return os.path.join(path, file_name)
+
+    def _export_data(
+        self,
+        export_type: DataExportType,
+        path: Optional[str] = None,
+        prefix: Optional[str] = None) -> str:
+        """Export data to disk with file name {prefix}{run_id}.{ext}.
+        Values for the export type, path and prefix are set in the qcodes
+        "dataset" config.
+
+        Args:
+            export_type: Data export type, e.g. DataExportType.NETCDF
+            path: Export path, defaults to value set in config
+            prefix: File prefix, e.g. "qcodes_", defaults to value set in config.
+
+        Returns:
+            str: Path file was saved to, returns None if no file was saved.
+        """
+        # Set defaults to values in config if the value was not set
+        # (defaults to None)
+        path = path or get_data_export_path()
+        prefix = prefix or get_data_export_prefix()
+
+        if DataExportType.NETCDF == export_type:
+            return self._export_as_netcdf(path=path, prefix=prefix)
+
+        elif DataExportType.CSV == export_type:
+            return self._export_as_csv(path=path, prefix=prefix)
+
+        else:
+            return ""
+
+    def export(
+        self,
+        export_type: Optional[Union[DataExportType, str]] = None,
+        path: Optional[str] = None,
+        prefix: Optional[str] = None) -> None:
+        """Export data to disk with file name {prefix}{run_id}.{ext}.
+        Values for the export type, path and prefix are set in the qcodes
+        "dataset" config.
+
+        Args:
+            export_type: Data
+                export type, e.g. "netcdf" or DataExportType.NETCDF,
+                defaults to value set config
+            path: Export path, defaults to value set in config
+            prefix: File prefix, e.g. "qcodes_", defaults to value set in config.
+
+        Raises:
+            ValueError: If the export data type is not specified, raise an error
+        """
+        export_type = get_data_export_type(export_type)
+
+        if export_type is None:
+            raise ValueError("No data export type specified. Please set the \
```

Comment:
thanks for pointing out the docstring discrepancy. I think I moved it here because I wanted the get_data_export_type function not to fail if it can't find an export type, but instead return None.